src/distributions/binomial.rs

```diff
@@ -41,16 +41,35 @@ impl Binomial {
     /// Construct a new `Binomial` with the given shape parameters `n` (number
     /// of trials) and `p` (probability of success).
     ///
-    /// Panics if `p <= 0` or `p >= 1`.
+    /// Panics if `p < 0` or `p > 1`.
     pub fn new(n: u64, p: f64) -> Binomial {
-        assert!(p > 0.0, "Binomial::new called with p <= 0");
-        assert!(p < 1.0, "Binomial::new called with p >= 1");
+        assert!(p >= 0.0, "Binomial::new called with p < 0");
+        assert!(p <= 1.0, "Binomial::new called with p > 1");
         Binomial { n, p }
     }
 }
 
 impl Distribution<u64> for Binomial {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u64 {
+        // Handle these values directly.
+        if self.p == 0.0 {
+            return 0;
+        } else if self.p == 1.0 {
+            return self.n;
+        }
+        
+        // For low n, it is faster to sample directly. For both methods,
+        // performance is independent of p. On Intel Haswell CPU this method
+        // appears to be faster for approx n < 300.
+        if self.n < 300 {
+            let mut result = 0;
+            let d = Bernoulli::new(self.p);
+            for _ in 0 .. self.n {
+                result += rng.sample(d) as usize;
```

Comment:
`usize` would make the results platform-depended. I think using `u32` is the best choice on all systems.

Reply:
We know `0 <= result < 300` so no overflow problems. Code will be different but results the same.

Reply:
I also think `u32` is a better choice here. IIRC there are platforms where `usize` has 64 bits but `u32` is faster.